### PR TITLE
Golang CI integration, linter fixes

### PR DESCRIPTION
**Pull request by AugustasV**
#### Summary
Added golang lint CI files, fixed some linter errors.

#### Pull request Link
  Fixes https://github.com/mattermost/mattermost-plugin-solar-lottery/issues/39

### CODE REVIEW

File: build/custom.mk [hunk 1]
```diff
@@ -1 +1 @@
-# Include custom targets and environment variables here
+# Include custome targets and environment variables here
```
Comment (hanzei) [inline build/custom.mk:1]: Why `custome` instead of `custom`?

File: Makefile [hunk 2]
```diff
@@ -44,81 +31,34 @@ all: check-style test dist
...
-	# mockgen -destination server/store/mock_store/mock_store.go github.com/mattermost/mattermost-plugin-solar-lottery/server/store Store
-endif
-
-## Generates mock golang interfaces for testing
-clean_mock:
-ifneq ($(HAS_SERVER),)
-	rm -rf ./server/solarlottery/mock_solarlottery
-	rm -rf ./server/utils/bot/mock_bot
-	rm -rf ./server/store/mock_store
-endif
```
Comment (hanzei) [inline Makefile:112]: These commands are still needed.

File: Makefile [hunk 3]
```diff
@@ -241,67 +172,3 @@ endif
...
-		kill -9 $$DELVE_PID ; \
-	fi
-
-	@echo "\nRestarting plugin via API"
-	$(eval TOKEN := $(shell curl -i -X POST $(MM_SERVICESETTINGS_SITEURL)/api/v4/users/login -d '{"login_id": "$(MM_ADMIN_USERNAME)", "password": "$(MM_ADMIN_PASSWORD)"}' | grep -o "Token: [0-9a-z]*" | cut -f2 -d' ' 2> /dev/null))
-	@curl -s -H "Authorization: Bearer $(TOKEN)" -X POST $(MM_SERVICESETTINGS_SITEURL)/api/v4/plugins/$(PLUGIN_ID)/disable > /dev/null && \
-		curl -s -H "Authorization: Bearer $(TOKEN)" -X POST $(MM_SERVICESETTINGS_SITEURL)/api/v4/plugins/$(PLUGIN_ID)/enable > /dev/null && \
-		echo "OK." || echo "Sorry, something went wrong. Check that MM_ADMIN_USERNAME and MM_ADMIN_PASSWORD env variables are set correctly."
-
-
```
Comment (hanzei) [inline Makefile:307]: Please leave these targets in the file. I'm aware that it's consistent with that is in the ticket, but it's not worth the cleanup in this PR.

File: Makefile [hunk 4]
```diff
@@ -44,81 +31,34 @@ all: check-style test dist
...
+	golangci-lint run ./...
 
 ## Builds the server, if it exists, including support for multiple architectures.
 .PHONY: server
 server:
 ifneq ($(HAS_SERVER),)
 	mkdir -p server/dist;
-	cd server && env GOOS=linux GOARCH=amd64 $(GOBUILD) -o dist/plugin-linux-amd64;
-	cd server && env GOOS=darwin GOARCH=amd64 $(GOBUILD) -o dist/plugin-darwin-amd64;
-	cd server && env GOOS=windows GOARCH=amd64 $(GOBUILD) -o dist/plugin-windows-amd64.exe;
```
Comment (hanzei) [inline Makefile:121]: My replacing `GOBUILD` a few needed environment variables also get dropped. I would suggest keeping these lines are they where and also keep the lines at the start of the file.

File: Makefile (hunk 3 above)
Comment (AugustasV) [reply Makefile:307]: Gotcha.

**Comment by AugustasV** [comment]
@hanzei tried to compare old Makefile and merge the missing pieces to the new one.
Waiting for comments, not sure I didn't reuse too much old code.

**Pull request merged**